db/model/generator.js

```diff
@@ -167,6 +167,10 @@ module.exports = function generator(seq, dataTypes) {
       beforeUpdate(inst /* , opts */) {
         const gtName = inst.generatorTemplate.name;
         const gtVersion = inst.generatorTemplate.version;
+        if (inst.changed('isActive')) {
```

Comment:
should this `if` also be checking whether the new value of `isActive` is `true`?

Reply:
No. If isActive was turned off we also need to call assignToCollector to set currentCollector to null.